### PR TITLE
fix the disk tag missing when NDM respawn

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
related issue: https://github.com/harvester/harvester/issues/2832

### CODE REVIEW

**Review by futuretea — APPROVED** [review]
LGTM

File: main.go [hunk 1]
```diff
@@ -165,6 +165,9 @@ func initLogs(opt *option.Option) {
 	}
 	logrus.SetOutput(os.Stdout)
 	logrus.Infof("Node Disk Manager %s is starting", version.FriendlyVersion())
+	logrus.Infof("Notable parameters are following:")
+	logrus.Infof("NameSpace: %s, ConcurrentOps: %d, RescanInterval: %d",
```
Comment (guangbochen) [inline main.go:169]:
```suggestion
	logrus.Infof("Namespace: %s, ConcurrentOps: %d, RescanInterval: %d",
```

**Review by guangbochen — APPROVED** [review]
small twist and everything else is LGTM.

File: main.go (hunk 1 above)
Comment (Vicente-Cheng) [reply main.go]: done, thanks!

**Comment by Vicente-Cheng** [comment]
hi all, fixed again with the auto-provision case.
Please help to review!

File: pkg/controller/blockdevice/scanner.go [hunk 2]
```diff
@@ -128,6 +128,11 @@ func (s *Scanner) scanBlockDevicesOnNode() error {
 	for _, device := range allDevices {
 		bd := device.bd
 		autoProvisioned := device.AutoProvisioned
+		/* do not need to enqueue again with ProvisionPhaseProvisioned disk */
+		if bd.Status.ProvisionPhase == diskv1.ProvisionPhaseProvisioned {
+			logrus.Infof("Skip the provisioned device: %s", bd.Name)
+			continue
+		}
```
Comment (bk201) [inline pkg/controller/blockdevice/scanner.go:135]: Not sure if this change caused issues to lines 140 ~143? (The `else if isDevPathChanged(oldBd, bd) {` clause)
IIRC, it's used to fix device path changes (e.g. swap disks when shutdown and boot)

Comment (Vicente-Cheng) [reply pkg/controller/blockdevice/scanner.go:135]: yap, that's a problem. We need this to work well when the host reboots.
Fixed this case by respecting disk tags from LH.

please review again!

**Review by bk201 — APPROVED** [review]
Thanks!

**Review by futuretea — APPROVED** [review]
LGTM

**Pull request merged**